### PR TITLE
#187: Implement periodic task for fetching tables

### PR DESCRIPTION
**Pull request by vladarefiev**
https://github.com/timelesslounge/timelessis/issues/187 : 
- Implemented periodic task for fetching tables from Poster
- Left puzzles for continue implementation and refactor

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#320](https://github.com/timelesslounge/timelessis/issues/320) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#320](https://github.com/timelesslounge/timelessis/issues/320) is assigned to @vryazanov[/z](https://www.0crat.com/u/vryazanov), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/23b018bf-4956-414a-8e9e-a1fed05c4d50); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: timeless/poster/tasks.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+            name=poster_table["name"], floor_id=poster_table["floor_id"]
+        ).first()
+
+        if table:
+            new_table = Table.merge_with_poster(table, poster_table)
+        else:
+            new_table = Table.create_by_poster(poster_table)
+
+        DB.session.add(new_table)
+        DB.session.flush()
```
Comment (vryazanov) [inline timeless/poster/tasks.py]: @vladarefiev need to `commit`, not `flush`

File: timeless/poster/tasks.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        table = DB.session(Table).query.filter_by(
+            name=poster_table["name"], floor_id=poster_table["floor_id"]
+        ).first()
+
+        if table:
+            new_table = Table.merge_with_poster(table, poster_table)
+        else:
+            new_table = Table.create_by_poster(poster_table)
+
+        DB.session.add(new_table)
```
Comment (vryazanov) [inline timeless/poster/tasks.py]: @vladarefiev need to `DB.merge` if table already exists

File: timeless/poster/tasks.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    """
+    Periodic task for fetching and saving tables from Poster
+    @todo #187:30min Set up scheduler for celery,
+     docs - http://docs.celeryproject.org/en/
+     latest/userguide/periodic-tasks.html#id5
+     Also should make small refactoring: celery.py should situated in
+     timelessis/celery.py not in timelessis/sync/celery.py
+    """
+    auth_data = PosterAuthData(
+        application_id=os.environ.get("poster_application_id"),
```
Comment (vryazanov) [inline timeless/poster/tasks.py]: @vladarefiev I suggest to move these vars to flask configs.

**Comment by vladarefiev** [comment]
@vryazanov thanks, pls take a look

**Comment by vryazanov** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@vladarefiev please check pylint in travis log

File: timeless/poster/tasks.py (hunk 3 above)
Comment (vryazanov) [reply timeless/poster/tasks.py]: @vladarefiev let's use vars from config, not from `os.environ`

**Comment by vladarefiev** [comment]
@emilianodellacasa I'm losing about .16 points in pylinter due to https://github.com/timelesslounge/timelessis/issues/168

And have no idea what can I fix in PR's code. May be fix another piece of code :)

**Comment by emilianodellacasa** [comment]
@vladarefiev you could fix this for example

timeless/poster/api.py:5:0: C0411: standard import "from urllib.parse import urljoin" should be placed before "import attr" (wrong-import-order)

I know it is not strictly yours but it's in the file you are modifiying. Please. :-)

**Comment by vladarefiev** [comment]
@emilianodellacasa 
pylint says
```
timeless/poster/models.py:5:0: R0903: Too few public methods (0/2) (too-few-public-methods)
```
what we actually have
```
class PosterSyncMixin:
    """Mixin with fields needed for data synchronization with Poster.
    """
    poster_id = DB.Column(DB.Integer)
    synchronized_on = DB.Column(DB.DateTime)
```

I think we definitely have to tune pylinter, because now it's shoots in our legs.

**Comment by vladarefiev** [comment]
@emilianodellacasa done

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job was finished in 22 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Job [#320](https://github.com/timelesslounge/timelessis/issues/320) is not in the agenda of @vryazanov[/z](https://www.0crat.com/u/vryazanov), can't inspect

**Comment by 0crat** [comment]
The job [#320](https://github.com/timelesslounge/timelessis/issues/320) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/timelesslounge/timelessis/issues/320#issuecomment-465736629))

@ypshenychka This is what I know about this job in [CF9LG1YCA](https://www.0crat.com/p/CF9LG1YCA), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#320](https://github.com/timelesslounge/timelessis/issues/320) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 26min
  * @vryazanov[/z](https://www.0crat.com/u/vryazanov) will receive 20 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/CF9LG1YCA?q=%7Bjob%3A%27gh%3Atimelesslounge%2Ftimelessis%23320%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@0crat quality good

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality is "good": +25 point(s) just awarded to @vryazanov[/z](https://www.0crat.com/u/vryazanov)

**Pull request merged**